packages/gatsby-theme-bodiless/src/dist/GitProvider.tsx

```
@@ -201,9 +202,36 @@ const getMenuOptions = (client: GitClient = defaultClient, context: any): TMenuO
   ];
 };
 
-const GitProvider: FC<Props> = ({ children, client }) => {
+const GitProvider: FC<Props> = ({ children, client = defaultClient }) => {
+  const [notifications, setNotifications] = useState([] as any);
   const context = useEditContext();
 
+  useNotify(notifications);
+
+  useEffect(() => {
+    const checkUpstreamChanges = async () => {
+      try {
+        const response = await client.getChanges();
+        if (response.status === 200) {
+          const isBranchOutdated = Object.keys(response.data).filter(branch => (
+            ['upstream', 'production'].includes(branch) && response.data[branch].commits.length
+          ));
+          if (isBranchOutdated) {
+            setNotifications([
+              ...notifications,
+              { id: 'upstreamChanges', message: 'You branch is outdated. Please pull remote changes.' },
+            ]);
+          }
+        }
+      } catch {
+        console.error('Fetching upstream changes failed');
```

maybe throw an error here `throw new Error('....')`

---

I think we should actually have a notification here that the attempt to check for upstream changes failed - please check with kevin at standup.

---

But you should def. throw an error as Ali suggests if you get a response status !== 200.

---

Per daily - handling of back-end errors will be moved to a gap ticket.
Will fail silently so far.